spec/controllers/projects_controller_spec.rb

```diff
@@ -1,14 +1,460 @@
 require 'rails_helper'
 
 RSpec.describe ProjectsController, type: :controller do
-  describe "GET index" do
-    login_admin
+  create_users
 
-    it "returns JSON with projects list" do
-      get :index, params: { format: 'json' }
+  # Admin specific behavior
+  context "Admin user" do
+    # create_users
+    before do
+      sign_in @admin
+    end
+
+    describe "GET index" do
+      it "sees all projects" do
+        expected_projects = [
+          create(:project, users: [@admin]),
+          create(:project, users: [@joe]),
+          create(:project, users: [@joe, @admin]),
+          create(:public_project),
+          create(:project, samples_data: [{ created_at: 1.year.ago }]),
+          create(:project, users: [@admin], samples_data: [{ created_at: 1.year.ago }])
+        ]
+        expected_projects.reverse!
```

Comment:
what are the `reverse!` for?

Reply:
By default, projects controller sorts projects by id in decreasing order. Probably I should change the expectation not to test order...
Do you have any other suggestions?

Reply:
Removed the expectations since they not core for the tests.